### DISCUSSION TITLE
Is it possible to use a different container runtime?

### DISCUSSION THREAD

**Discussion by incognick** [testcontainers/testcontainers-go#2837] [Q&A]
I see that testcontainers requires a docker compatible API. Is it possible to pass in a different runtime?  Such as `runc` or `containerd` ?  Forgive me if I'm confusing different topics as I'm still learning all the possibilities. Thanks!

**Comment by mdelapenya** [comment]
Hi @incognick thanks for reaching out. Testcontainers should work with any container runtime that is 100% compatible with the Docker APIs. There are some runtimes that need special configuration, which we describe in our docs, but we do not have anything for runc or containerd.

OTOH we are working in a more transparent support: you activate a Docker context with the docker socket, and Testcontainers should work out of the box. This is WIP, please see our work in #2810
